Packs/Code42/Integrations/Code42/Code42.yml

```diff
@@ -234,6 +239,59 @@ script:
     - contextPath: File.Hostname
       description: The hostname where the file event was captured.
       type: string
+  - arguments:
+    - default: false
+      description: Add results to context at 'Code42.FileEvents'. If 'false', the search will only display results as a markdown table.
+      isArray: false
+      name: add-to-context
+      required: false
+      secret: false
+      defaultValue: false
+    - default: false
+      description: Raw JSON file event query to be used for search.
+      isArray: false
+      name: json
+      required: false
+      secret: false
+    - default: false
+      description: The number of file events to return. Defaults to 50.
+      isArray: false
+      name: results
+      required: false
+      secret: false
+      defaultValue: 50
+    - default: false
+      defaultValue: "1"
+      description: Filter results by minimum risk score.
+      isArray: false
+      name: min_risk_score
+      required: false
+      secret: false
+    - default: false
+      description: MD5 or SHA256 hash of the file to search for.
+      isArray: false
+      name: hash
+      required: false
+      secret: false
+    - default: false
+      description: Username to search for.
+      isArray: false
+      name: username
+      required: false
+      secret: false
+    - default: false
+      description: Hostname to search for.
+      isArray: false
+      name: hostname
+      required: false
+      secret: false
+    deprecated: false
+    description: Search for Code42 Incydr File Events
+    execution: false
+    name: code42-file-events-search
+    outputs:
+      - contextPath: Code42.FileEvents
+        description: List of file events returned from searches.
```

Comment:
Code42 file events are quite large with many nested fields, with new fields being added frequently. Is enumerating them all here (and keeping them up-to-date) a requirement?

An example event: 
```
{
      "@timestamp": "2022-12-09T19:03:11.954000Z",
      "destination": {
        "accountName": null,
        "accountType": null,
        "category": "Device",
        "domains": [],
        "email": {
          "recipients": null,
          "subject": null
        },
        "ip": null,
        "name": "Unmonitored Device",
        "operatingSystem": null,
        "printJobName": null,
        "printerName": null,
        "printedFilesBackupPath": null,
        "privateIp": [],
        "removableMedia": {
          "busType": null,
          "capacity": null,
          "mediaName": null,
          "name": null,
          "partitionId": [],
          "serialNumber": null,
          "vendor": null,
          "volumeName": []
        },
        "tabs": [],
        "user": {
          "email": []
        }
      },
      "event": {
        "action": "file-downloaded",
        "id": "0106499c-89d0-4809-bef8-f647a6059982_EPS",
        "ingested": "2022-12-09T19:03:12.329000Z",
        "inserted": "2022-12-09T19:51:57.374625Z",
        "observer": "Salesforce",
        "relatedEvents": null,
        "shareType": [],
        "vector": "DOWNLOADED",
        "detectorDisplayName": null
      },
      "file": {
        "acquiredFrom": [],
        "category": "Spreadsheet",
        "categoryByBytes": null,
        "categoryByExtension": "Spreadsheet",
        "changeType": "NONE",
        "classifications": [],
        "cloudDriveId": null,
        "created": "2022-12-09T19:03:08Z",
        "directory": null,
        "directoryId": [],
        "hash": {
          "md5": null,
          "md5Error": null,
          "sha256": null,
          "sha256Error": null
        },
        "id": null,
        "mimeTypeByBytes": null,
        "mimeTypeByExtension": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
        "modified": null,
        "name": "New Contacts & Accounts Report-2022-12-09-11-03-08.xlsx",
        "owner": null,
        "sizeInBytes": null,
        "url": null,
        "originalName": null,
        "originalDirectory": null,
        "mimeType": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
      },
      "process": {
        "executable": null,
        "owner": null
      },
      "report": {
        "count": 20,
        "description": null,
        "headers": [
          "SALUTATION",
          "FIRST_NAME",
          "LAST_NAME",
          "TITLE",
          "ACCOUNT.NAME",
          "ADDRESS2_STREET",
          "ADDRESS2_CITY",
          "ADDRESS2_STATE",
          "ADDRESS2_ZIP",
          "ADDRESS2_COUNTRY",
          "PHONE1",
          "PHONE2",
          "PHONE3",
          "EMAIL",
          "OWNER_FULL_NAME"
        ],
        "id": "00O5f000002mwE0",
        "name": "New Contacts & Accounts Report",
        "type": "REPORT_TYPE_SAVED"
      },
      "risk": {
        "indicators": [
          {
            "name": "Download to unmonitored device from corporate Salesforce",
            "weight": 6,
            "id": "Download to unmonitored device from corporate Salesforce"
          },
          {
            "name": "Spreadsheet",
            "weight": 0,
            "id": "Spreadsheet"
          }
        ],
        "score": 6,
        "severity": "MODERATE",
        "trustReason": null,
        "trusted": false,
        "preventativeControl": null,
        "untrustedValues": {
          "accountNames": [],
          "domains": [],
          "gitRepositoryUris": [],
          "slackWorkspaces": [],
          "urlPaths": []
        }
      },
      "source": {
        "accountName": null,
        "accountType": null,
        "category": "CRM Tools",
        "domain": null,
        "domains": [],
        "email": {
          "from": null,
          "sender": null
        },
        "ip": null,
        "name": "Salesforce",
        "operatingSystem": null,
        "privateIp": [],
        "removableMedia": {
          "busType": null,
          "capacity": null,
          "mediaName": null,
          "name": null,
          "partitionId": [],
          "serialNumber": null,
          "vendor": null,
          "volumeName": []
        },
        "tabs": [],
        "user": {
          "email": []
        }
      },
      "user": {
        "deviceUid": null,
        "email": "jim.harper@example.com",
        "id": "1004440923549707297"
      },
      "git": {
        "eventId": null,
        "lastCommitHash": null,
        "repositoryEmail": null,
        "repositoryEndpointPath": null,
        "repositoryUri": null,
        "repositoryUser": null
      }
```

Reply:
It is not required to add them all, but it is easier to use the fields from the playbooks if they are mentioned as outputs in yml. You can add to the outputs only the first-level fields, it can be ok as well. For example add only: destination,event,file,process,risk,source,user, git fields to the outputs.

Reply:
Ok, updated to include all the top-level fields.